### PR TITLE
feat(api-client): environment table editor

### LINKED ISSUE #2704: (Scalar API Client): rethink environment page
### What happens?

Currently our environment page is purely a JSON editor, we should move to a table approach.

### What did you expect to happen?

a

### How can we reproduce the issue?

a

### OpenAPI Document

_No response_

### PR DESCRIPTION
**Pull request by antlio**
this pr is a first proposal to close #2704 to move from editor only to table first for the environment page.

https://github.com/user-attachments/assets/b239bb5e-54a1-4edf-b5a6-003e5c70fff9

### CODE REVIEW

**Review by hanspagel — CHANGES_REQUESTED** [review]
Really cool 🙌 

I get some weird behaviour when I delete a few variables and add a new one. Old variables pop up again then.

https://github.com/user-attachments/assets/361e3f05-2a45-4bc5-92e8-bd37a3c3ac59

https://github.com/user-attachments/assets/d59539ff-9f1b-42ab-bd2b-cad20bfcb65d

https://github.com/user-attachments/assets/5ae77218-295e-4360-924d-417ba7e6c1a6

**Review by marclave — APPROVED** [review]
can merge once the entities work get's finished ✨

**Comment by hanspagel** [comment]
Blocked by #2931

**Comment by amritk** [comment]
This one might be easier to just pick the files instead of rebasing

**Comment by antlio** [comment]
> This one might be easier to just pick the files instead of rebasing

I started this again, will push code asap in that same branch

**Comment by hanspagel** [comment]
OMG, can’t believe we’re so close!

Just found some strange behaviour when two keys start the same. Did you try to attach uuids to all entries?

https://github.com/user-attachments/assets/30ba15bb-2e1c-4792-88a8-632b68cc717f

File: packages/components/src/components/ScalarIcon/icons/icons.ts
Comment (hwkr) [inline packages/components/src/components/ScalarIcon/icons/icons.ts:1]: Is this replacing the `index.ts` file?

Comment (antlio) [reply packages/components/src/components/ScalarIcon/icons/icons.ts:1]: good catch i guess rebase issue here 🤔 will fix!

**Comment by antlio** [comment]
closing this for now - will get back to it with a better approach later on

**Pull request closed without merging**